src/core/thread/mle_router.cpp

```diff
@@ -584,6 +585,14 @@ otError MleRouter::SendAdvertisement(void)
         break;
     }
 
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+    if (GetNetif().GetTimeSync().IsTimeSyncRequired())
```

Comment:
Do we need to make sure we send advertisements without Time IEs to retain backwards compatibility?

Thinking more about this, is there a strong advantage to piggybacking Time IEs in MLE Advertisement?  Couple issues I see:
1. Breaks backwards compatibility.
2. Increases MLE Advertisement size, which can cause fragmentation issues later on.

What if we use a separate message type for propagating Time IEs after attached?

Reply:
I think the only benefit to reuse MLE Advertisement is that it could save some power consumption:
* If time sync period is 30s, all regular MLE Advertisement will include Time IE, so no extra messages power cost, but yes, it will break backward compatibility;
* If time sync period is more than 30s, for example 90s, so only one in every three MLE Advertisement will include Time IE.

I also like the idea to use a new message type for Time sync, so there's no effect to the regular MLE Advertisement, the only concern is that it may increase more power consumption. Which way do you prefer?

Reply:
I'm not sure we care too much about power consumption given that devices sending MLE Advertisements will also be rx-on-when-idle.  For that reason, I think it is better to start with having a new message for the time sync.

Reply:
Ok, Done. A separate message type is introduced for time synchronization.